includes/classes/order.php

```diff
@@ -383,6 +383,24 @@ function cart() {
 
     }
 
+    // -----
+    // A shipping-module's 'code', if present in the session, **must** contain a '_' character, separating
+    // the shipping module's name from the selected method, e.g. 'module_method'.  That '_' cannot be the first
+    // character of the 'code' value.
+    //
+    // If that's not the case, issue a PHP Notice and reset the shipping to its unselected state.
+    //
+    if (empty($_SESSION['shipping']['id'])) {
+        $shipping_module_code = '';
+    } else {
+        if (!strpos((string)$_SESSION['shipping']['id'], '_')) {
+            trigger_error('Malformed value for session-based shipping module; customer will need to re-select: ' . json_encode($_SESSION['shipping']), E_USER_NOTICE);
+            unset($_SESSION['shipping']);
+            $shipping_module_code = '';
+        } else {
+            $shipping_module_code = $_SESSION['shipping']['id'];
+        }
+    }
```

Comment:
This nested if/else/if/else could be made much simpler and shorter by just setting the string to blank beforehand (replace the outer if/else) and then only keep the inner if/else and remove the then-redundant `= ''` inside it.